### PR TITLE
Teleport rules optimization

### PR DESCRIPTION
**Pull request by ssyno**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
Towards: https://github.com/giantswarm/giantswarm/issues/30312
This PR 
- fixes teleport related resource alerts

### Checklist

- [ ] Update CHANGELOG.md
- [x] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [x] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml [hunk 1]
```diff
@@ -16,17 +16,11 @@ spec:
       expr: |
         count (
           kube_secret_created{secret=~".*-teleport-join-token"}
-        ) by (cluster_id, installation)
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: Can you add the by (cluster_id, installation) back ? Those are needed for milir

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: Why are they différent ?

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: That's why we had the relabel in place in the second expression

Comment (ssyno) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: By summing them by cluster_id/installation you have 1 cluster of each 
```
        sum (
          label_replace(
            capi_cluster_status_phase{phase="Provisioned"},
            "cluster_id",
            "$1",
            "name",
            "(.*)"
          )
        ) by (cluster_id, installation)
```
image

and we need the total of them to compare with 
```
        count (
          kube_configmap_info{configmap=~".*-teleport-kube-agent-config"}
        ) by (cluster_id, installation)
```
image

Comment (ssyno) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: We can can keep the label_replace but we have to get rid of the `by (cluster_id, installation)` part

Comment (ssyno) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: or we can add an additional sum on top

**Comment by QuentinBisson** [comment]
We do need some basic things for alert routing. Let's figure them out next week

File: helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml (hunk 1 above)
Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: You might need to use something like this :( https://github.com/giantswarm/prometheus-rules/blob/3e1587b2ee57abcd98dbf07e193b50b481444d23/helm/prometheus-rules/templates/recording-rules/grafana-cloud.rules.yml#L8

unless @hervenicol can think of a better solution?

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: After some testing on grizzly, I would replace the configmap part with:

```promql
count (
    label_replace(
      kube_configmap_info{configmap=~".*-teleport-kube-agent-config"}, 
      "cluster_id",
      "$1",
      "configmap", 
      "(.*)-teleport-kube-agent-config"
    )
) by (cluster_id, installation, provider, pipeline) 
!= sum (
    capi_cluster_status_phase{phase="Provisioned"}
) by (cluster_id, installation, provider, pipeline)
```

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: and the secret one with:


```promql
count (
    label_replace(
      kube_secret_created{secret=~".*-teleport-join-token"}, 
      "cluster_id",
      "$1",
      "secret", 
      "(.*)-teleport-join-token"
    )
) by (cluster_id, installation, provider, pipeline)
!= sum (
    capi_cluster_status_phase{phase="Provisioned"}
) by (cluster_id, installation, provider, pipeline)
```

Comment (ssyno) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: nice! LGTM

Comment (ssyno) [reply helm/prometheus-rules/templates/alerting-rules/teleport.rules.yml:19]: updating the queried on rules

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuentinBisson** [comment]
I think you just need to fix the tests :)

**Comment by ssyno** [comment]
> I think you just need to fix the tests :)

fixed :)

**Comment by QuentinBisson** [comment]
Great job

**Pull request merged**